packages/build/src/plugins_core/build_command.js

```diff
@@ -40,7 +40,7 @@ const coreCommand = async function ({
   } catch (newError) {
     handleBuildCommandOutput(newError, logs)
     addErrorInfo(newError, { type: 'buildCommand', location: { buildCommand, buildCommandOrigin, configPath } })
-    return { newError }
+    throw newError
```

Comment:
This is caught by the calling function, which returns it as `{ newError }` so this change should have the same behavior.
However, it makes it clear that core commands are expected to throw errors, not return them.